### ISSUE TITLE
with the changes to physics_body_2d, I cant find a replacement for revert_motion on a KinematicBody2D

### ISSUE THREAD

**Issue by rubenjavier** [godotengine/godot#9397]
**Operating system or device - Godot version:**
Linux Ubuntu 64bit commit 36548a3fa58f00a3ef301f169f237b5641bdf635

**Issue description:**
cant find revert_motion on the physics_body_2d code anymore, what is the replacement to avoid the ethernal sliding on slopes?

**Steps to reproduce:**
test code from a couple of days ago with revert_motion() to avoid sliding on slopes, it wont work

**Comment by kubecz3k** [comment]
Seems like an oversight. @reduz can it be implemented the same way how it's done in KinematicBody 3D? https://github.com/godotengine/godot/blob/master/scene/3d/physics_body.cpp#L911

**Comment by rubenjavier** [comment]
@kubecz3k @reduz 
Maybe this is also a good oportunity to fix a couple of lines inside physics_body_2d
inside Dictionary:
```
Dictionary KinematicBody2D::_move(const Vector2 &p_motion)
```
normal is wrong (it's saving .collision), it needs to be:
(also, please add collider_velocity to the Dictionary as follows)
```
   d["normal"] = col.normal;
   d["collider_velocity"] = col.collider_vel;
```

Thanks

**Comment by akien-mga** [comment]
We have now entered release freeze for Godot 3.0 and want to focus only on release critical issues for that milestone. Therefore, we're moving this issue to the 3.1 milestone, though a fix may be made available for a 3.0.x maintenance release after it has been tested in the master branch during 3.1 development. If you consider that this issue is critical enough to warrant blocking the 3.0 release until fixed, please comment so that we can assess it more in-depth.

**Comment by reduz** [comment]
You just move back your object using the travel property of the collider

```
var c = move_and_collide(blabla)
if (c):
     position -= c.travel
```
